### PR TITLE
fabtests/benchmarks: Add synapseai support

### PR DESCRIPTION
**Pull request by imasuari-habana**
The following changes adds synapaseai support to fabtests' benchmark tests.
Choosing iface value of FI_HMEM_SYNAPSEAI will implicitly set option FT_OPT_REG_DMABUF_MR.

### CODE REVIEW

**Comment by nbeer-habana** [comment]
@j-xiong We can't see the CI failures, can you help?

**Comment by j-xiong** [comment]
The build stage was interrupted for unknown reason. I restarted CI.

**Comment by shijin-aws** [comment]
@imasuari-habana Does/did HCCL onboard FI_MR_DMABUF ?

File: fabtests/common/hmem_synapseai.c [hunk 1]
```diff
@@ -0,0 +1,446 @@
...
+	return stream_synchronize(synapseai_stream_handle);
+}
+
+int ft_synapseai_get_dmabuf_fd(void *buf, size_t len, int *dmabuf_fd, uint64_t *dmabuf_offset)
+{
+	scal_pool_handle_t mpHandle;
+	scal_memory_pool_infoV2 mpInfo;
+	scal_handle_t a = 0;
+
+	if (synapseai_ops.scal_get_handle_from_fd(device_fd, &a) != SCAL_SUCCESS) {
```
Comment (shijin-aws) [inline fabtests/common/hmem_synapseai.c:366]: Should we update https://github.com/ofiwg/libfabric/blob/main/src/hmem_synapseai.c#L192 ?

Comment (imasuari-habana) [reply fabtests/common/hmem_synapseai.c:366]: We consider moving hcl code to use FI_MR_DMABUFF registration method. so currently it should not be updated

**Comment by imasuari-habana** [comment]
@j-xiong there are still 2 failed jobs in CI which I can not open. can you help again?

**Comment by j-xiong** [comment]
The failure are with Intel MPI test over tcp which should be unrelated to the changes here.

**Review by j-xiong — APPROVED** [review]

**Pull request merged**